src/Chainweb/Pact/PactService.hs

```diff
@@ -648,51 +669,51 @@ rewindTo mb act = do
 
     failNonGenesisOnEmptyDb = fail "impossible: playing non-genesis block to empty DB"
 
-    playFork cp bhdb payloadDb newH parentHash (lastBlockHeight, lastHash) =
-      flip catches exHandlers $ do
-          parentHeader <- liftIO $ lookupM bhdb parentHash
-          lastHeader <- liftIO $ lookupM bhdb lastHash
-          (!_, _, newBlocks) <-
-              liftIO $ collectForkBlocks bhdb lastHeader parentHeader
-          -- play fork blocks
-          V.mapM_ (fastForward payloadDb) newBlocks
-          -- play new block
-          restoreCheckpointer (Just (newH, parentHash)) >>= act
+    playFork cp bhdb payloadDb newH parentHash (lastBlockHeight, lastHash) = do
+        parentHeader <- liftIO $ lookupM bhdb parentHash
+        lastHeader <- findValidParent lastBlockHeight lastHash
+
+        (!_, _, newBlocks) <-
+            liftIO $ collectForkBlocks bhdb lastHeader parentHeader
```

Comment:
@gregorycollins one question we had from last night is this line here and what we're doing by forcing the db ref, and only retrieving the new blocks. Could either you or @emmanueldenloye annotate this with some docs?

Reply:
Would you rather it be done now or in follow up PR?

Reply:
Actually, I have taken a look. I feel pretty confident about removing the bang.